### PR TITLE
change the HOME for git commands

### PR DESCRIPTION
**Pull request by gaborbernat**
this avoids using the users configuration, which can contain e.g. gpg signing with password which would break this tests

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -39,6 +39,11 @@ def get(self):
     yield TmpdirFactory()
 
 
+@pytest.fixture()
+def setup_git(monkeypatch, tmpdir):
+    # we should not use user configuration
+    monkeypatch.setenv('HOME', tmpdir)
```
Comment (asottile) [inline tests/conftest.py]: we had the same situation in pre-commit-hooks and I think this is the wrong solution as it has the potential to *hide* problems with different configurations.  We instead opted to pass `--no-gpg-sign` to any `git commit`s under test: https://github.com/pre-commit/pre-commit-hooks/pull/121

Comment (asottile) [reply tests/conftest.py]: this fixture also doesn't scale (adding it to *literally every test*) -- perhaps instead an `autouse` fixture would have been more appropriate

File: tests/repository_test.py [hunk 2]
```diff
@@ -41,50 +43,49 @@ def _norm_out(b):
...
-    repo = Repository.create(config, store)
-    hook_dict, = [
-        hook for repo_hook_id, hook in repo.hooks if repo_hook_id == hook_id
-    ]
-    ret = repo.run_hook(hook_dict, args)
-    assert ret[0] == expected_return_code
-    assert _norm_out(ret[1]) == expected
+@pytest.fixture(name='test_hook_repo')
+def test_hook_repo_fixture(tempdir_factory, store, setup_git):
+    def do(repo_path,
```
Comment (asottile) [inline tests/repository_test.py]: why this change?

File: testing/util.py [hunk 3]
```diff
@@ -42,6 +42,16 @@ def cmd_output_mocked_pre_commit_home(*args, **kwargs):
...
 
+skipif_cant_run_go = pytest.mark.skipif(
+    parse_shebang.find_executable('go') is None,
+    reason='go isn\'t installed or can\'t be found',
+)
+
+skipif_cant_run_bash = pytest.mark.skipif(
+    parse_shebang.find_executable('bash') is None,
+    reason='bash isn\'t installed or can\'t be found',
+)
```
Comment (asottile) [inline testing/util.py]: without bash, most of the testsuite is useless -- pre-commit tests lots of the framework functionality with script hooks as they are _fast_.  I don't think I can accept a PR which adds skips for `bash`.  As `bash` is necessarily installed on all the platforms we support (it's necessary for `git` itself to function) this also seems inappropriate.

File: tests/repository_test.py [hunk 4]
```diff
@@ -616,7 +616,10 @@ def test_invalidated_virtualenv(tempdir_factory, store):
     paths = [
         os.path.join(libdir, p) for p in ('site.py', 'site.pyc', '__pycache__')
     ]
-    cmd_output('rm', '-rf', *paths)
+
+    for path in paths:
+        if os.path.exists(path):
+            os.rmdir(path)
```
Comment (asottile) [inline tests/repository_test.py]: 👍

**Comment by gaborbernat** [comment]
@asottile I did not knew bash is required. I'm personally not comfortable knowing  this to add this project to tox; as you make it at best cumbersome for people on Windows to develop; now they need to install some kind of bash beside just Python.

**Comment by asottile** [comment]
It's not required for runtime, just very important for the test suite. Also if you're running git on windows, you're running bash ;)

**Comment by gaborbernat** [comment]
@asottile I'm using cmder which includes bash, however the test fails as it tries to find ``/usr/bin/bash`` which is not here; any ideas?

**Comment by asottile** [comment]
Can you produce the output of the test as well as `which bash` and `echo $PATH`?

**Comment by gaborbernat** [comment]
```
C:\Users\berna\git\tox (speed)
λ env | grep PATH
GIT_LFS_PATH=C:\Program Files\Git LFS
HOMEPATH=\Users\berna
PATH=/c/Program Files/cmder/bin:/c/Program Files/cmder/vendor/conemu-maximus5/ConEmu/Scripts:/c/Program Files/cmder/vendor/conemu-maximus5:/c/Program Files/cmder/vendor/conemu-maximus5/ConEmu:/c/ProgramData/Oracle/Java/javapath:/c/Program Files (x86)/Intel/iCLS Client:/c/Program Files/Intel/iCLS Client:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Program Files (x86)/Intel/Intel(R) Management Engine Components/DAL:/c/Program Files/Intel/Intel(R) Management Engine Components/DAL:/c/Program Files (x86)/Intel/Intel(R) Management Engine Components/IPT:/c/Program Files/Intel/Intel(R) Management Engine Components/IPT:/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cmd:/c/WINDOWS/system32:/c/WINDOWS:/c/WINDOWS/System32/Wbem:/c/WINDOWS/System32/WindowsPowerShell/v1.0:/c/Program Files/Git LFS:/c/Program Files/PuTTY:/c/Program Files/nodejs:/c/HashiCorp/Vagrant/bin:/:/c/ProgramData/Anaconda3:/c/ProgramData/Anaconda3/Scripts:/c/Users/berna/AppData/Local/conda/conda/envs:/c/Users/berna/AppData/Local/Microsoft/WindowsApps:/c/Program Files (x86)/Microsoft VS Code/bin:/c/Users/berna/AppData/Roaming/npm:/c/Users/berna/AppData/Local/Google/Cloud SDK/google-cloud-sdk/bin:/usr/bin:/usr/share/vim/vim74:/c/Program Files/cmder
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
VBOX_MSI_INSTALL_PATH=C:\Program Files\Oracle\VirtualBox\

C:\Users\berna\git\tox (speed)
λ which bash
/c/Windows/system32/bash
```

File: tests/conftest.py (hunk 1 above)
Comment (gaborbernat) [reply tests/conftest.py]: I see
